src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/JsonConverterCollectionExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Razor.ProjectEngineHost.Serialization;
+
+internal static class JsonConverterCollectionExtensions
+{
+    private static readonly IReadOnlyList<JsonConverter> ProjectSerializerConverters = new List<JsonConverter>()
```

Comment:
Make this a `List<JsonConverter>` and you can do an allocation-free `foreach` below.